Qadir__Work/JavaScript/Assignment14/Question1/Qadir__Assignment14__1.html

```diff
@@ -0,0 +1,64 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Inheritence</title>
+  </head>
+  <body>
+      <!-- language attribute is optional -->
+    <script language="javascript">
+        // 1. class Name is Always start with Captial Letter not Small
+        // class student -- class Student
+        class Student { 
+          constructor(RollNo, Name) {
+              this.RollNo = RollNo;
+              this.Name = Name;
+          }
+
+          display() {
+              document.write("<br>Inside Display");
+              document.write("<br>Roll No " + this.RollNo);
+              document.write("<br>Name " + this.Name);
+          }
+        }
+
+        //Marks is a derived class/child class
+        class Marks extends Student {
+          //Construcor Overrriding
+          // 2. if we are overriding class variable we need to declare super() inside the constructor
+          // Also parameter name and class variable name is always in small case
+          constructor(RollNo, Name, Marks) {
+              super(RollNo, Name);
+              this.RollNo = RollNo;
+              this.Name = Name;
```

Comment:
no need to initialize rollno and name as they will be intitalized in the super class constructor

Reply:
Ok Mam.